juju/juju
# [JUJU-1910] Use base api params for upgrade-series and charm origin

Introduce a Channel struct to the series package to handle os version and risk. We now use a base struct over the wire for deploy, add machine, upgrade series etc. For now, the risk is ignore as we still use series internally, but the over the wire format can support it if we decide to use it later. The 2.9 api supports old and new api params. For 3.0, we will transition to not use series over the wire at all.

Alias the set-series command to set-base and upgrade-series to upgrade-machine.

## Checklist


## QA steps

deploy test bundle
```
applications:
  juju-qa-test:
    charm: juju-qa-test
    channel: stable
    num_units: 2
  nrpe:
    charm: cs:nrpe-64
    channel: stable
  ntp:
    charm: cs:ntp-40
    channel: stable
    series: focal
  telegraf:
    charm: cs:telegraf-37
    channel: stable
  ubuntu:
    charm: ubuntu
    series: focal
    num_units: 2
      #    to:
      #      - "lxd:new"
relations:
- - ntp:juju-info
  - juju-qa-test:juju-info
- - ntp:juju-info
  - ubuntu:juju-info
- - nrpe:general-info
  - ubuntu:juju-info
- - telegraf:juju-info
  - ubuntu:juju-info
```
do it again to check bundle diff works
juju status --format yaml
```
juju set-base juju-qa-test jammy
ERROR charm "juju-qa-test" does not support jammy, force not used
```

```
juju upgrade-machine 4 prepare jammy
WARNING: This command will mark machine "4" as being upgraded to series "jammy".
This operation cannot be reverted or canceled once started.
Units running on the machine will also be upgraded. These units include:
  - nrpe/0
  - ntp/1
  - telegraf/0
  - ubuntu/1

Leadership for the following applications will be pinned and not
subject to change until the "complete" command is run:
  - nrpe
  - ntp
  - telegraf
  - ubuntu

Continue [y/N]?
```
